src/sampling.rkt

```diff
@@ -119,6 +123,63 @@
     (timeline-push! 'method "random")
     (cons (λ () (map random-generate reprs)) (hash 'unknown 1.0))]))
 
+(define bool-discretization
+  (discretization identity
+                  (lambda (x y) (if (eq? x y) 0 1))))
```

Comment:
I see, we want the boolean result to have the same values at endpoints? Maybe we want to make sure that they are both true. Probably, it is not a necessary rewrite, but anyways

Reply:
Yeah this is saying the `[true, true]` and `[false, false]` intervals have converged, while the `[false, true]` interval hasn't. This is used for boolean outputs (like preconditions), we want to stop early if the precondition is `[false, false]`.